src/main/java/de/uhd/ifi/se/decision/management/jira/extraction/versioncontrol/GitDiffedCodeExtractionManager.java

```diff
@@ -0,0 +1,329 @@
+ * following syntax inside a source code comment:
+ * """
+ *
+ * @decKnowledgeTag: knowledge summary text
+ *
+ * knowledge description text after empty line
+ *
+ *
+ * """
+ *
+ * where [decKnowledgeTag] belongs to set of know Knowledge Types,
+ * for example issue, alternative, decision etc.
+ * and where empty two lines denote the end of the decision element,
+ * Observation of another tag ends the element too.
+ * Comment end ends also the decision element.
+ *
+ * This class will:
+ * 1) fetch with gitClient necessary files from code repository,
+ * 2) delegate comment extraction to specialized classes and
+ * 3) extract evolution of knowledge elements in files.
+ */
+public class GitDiffedCodeExtractionManager {
+	private static final String OLD_FILE_SYMBOL_PREPENDER = "~";
+	/*
+	 * @issue: Modified files may add, modify or delete rationale.
+	 * For extraction of rationale on modified files their contents
+	 * from the base and changed versions are required.
+	 * How should both versions of affected files be accessed?
+	 *
+	 * @decision: Use two gitClientImpl instances each checked out
+	 * at different commit of the diff range!
+	 * @pro: no additional checkouts need to be performed by the gitClientImpl
+	 * @con: requires on more gitClientImpl object in the memory
+	 *
+	 * @alternative: Switch between commits using one gitClientImpl
+	 * instance!
+	 * @con: frequent checkouts take time
+	 * @con: requires implementation of another special mode in the
+	 * client
+	 *
+	 */
+
+	private final GitClient gitClientCheckedOutAtDiffStart;
+	private final GitClient gitClientCheckedOutAtDiffEnd;
+	private final Map<DiffEntry, EditList> diffEntries;
+
+	private static final Logger LOGGER = LoggerFactory.getLogger(GitDiffedCodeExtractionManager.class);
+
+	// may include null values for keys!
+	private Map<DiffEntry, CodeExtractionResult> results = new HashMap<>();
+
+	public GitDiffedCodeExtractionManager(Map<DiffEntry, EditList> diffEntries
+			, GitClient GitClientAtDiffEnd
+			, GitClient GitClientAtDiffStart) {
+		gitClientCheckedOutAtDiffStart = GitClientAtDiffStart;
+		gitClientCheckedOutAtDiffEnd = GitClientAtDiffEnd;
+		this.diffEntries = diffEntries;
+		processEntries();
+	}
+
+
+	public List<DecisionKnowledgeElement> getNewDecisionKnowledgeElements() {
+		return getNewOrOldDecisionKnowledgeElements(true);
+	}
+
+	public List<DecisionKnowledgeElement> getOldDecisionKnowledgeElements() {
+		return getNewOrOldDecisionKnowledgeElements(false);
+	}
+
+	private List<DecisionKnowledgeElement> getNewOrOldDecisionKnowledgeElements(boolean getNew) {
+		List<DecisionKnowledgeElement> resultValues = new ArrayList<>();
+
+		if (results.size() > 0) {
+			for (Map.Entry<DiffEntry, CodeExtractionResult> dEntry
+					: results.entrySet()) {
+				String newPath;
+				if (getNew) {
+					newPath = dEntry.getKey().getNewPath();
+				} else {
+					newPath = OLD_FILE_SYMBOL_PREPENDER + dEntry.getKey().getNewPath();
```

Comment:
What's the different to getOldPath()?

Reply:
oh oh, it should be getOldPath in the else block, thanks!